### ISSUE TITLE
.fpl format - entries.unk0 and primary_values

### ISSUE THREAD

**Issue by AkaBlur** [rr-/fpl_reader#2]
Hi, I'm not sure if this repo is still maintained, but I have figured out some additional information about the .fpl format.
I'm working on a similar tool for playlists and discovered the RE for the `.fpl`-format here.

https://github.com/rr-/fpl_reader/blob/d4099a597a3eb2fe18f410334639e3189eac24af/fpl-format.md?plain=1#L199-L203
The `unk0` mentioned here is a 32-bit value which represents the number of values inside the `entries.primary_values`.

Taking `unk0` as `primary_value_count` for the calculation for the size of `primary_values` is then:
4 * `primary_value_count` bytes (not `primary_key_count`)

Basically when you have a duplicate value inside the `primary_keys` field, the duplicate key itself is not stored. But inside the `primary_values` all keys have a value assigned. To calculate the offset correctly `unk0` closes the gap for this.

The example for the discontiguous keys inside [fpl-format.md](https://github.com/rr-/fpl_reader/blob/master/fpl-format.md) also shows this but the calculation for the offset wouldn't be correct then.

I have confirmed this layout for **foobar v2.0**
